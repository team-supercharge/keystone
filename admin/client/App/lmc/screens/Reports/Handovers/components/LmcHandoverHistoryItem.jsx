import React, { Component } from 'react'
import PropTypes from 'prop-types'
import AnimateHeight from 'react-animate-height'
import LmcHandoverResidentItem from './LmcHandoverResidentItem.jsx'
import LmcHandoverNotes from './LmcHandoverNotes.jsx'

export default class LmcHandoverHistoryItem extends Component {
    render () {
        const { logsByResident, notes, seenBy } = this.props.handover
        return (
            <div style={styles.dataContainer}>
                <div style={styles.leftContainer}>
                    { logsByResident.map((logGroup, i) => {
                        return (
                            <div key={i}>
                                <LmcHandoverResidentItem
                                    data={logGroup}
                                />
                            </div>
                        )
                    })}
                </div>
                <div style={styles.rightContainer}>
                    <LmcHandoverNotes
                        notes={notes}
                    />
                </div>
            </div>
        )
    }
}

const styles = {
    dataContainer: {
        display: 'flex',
        flexDirection: 'row',
        width: '100%'
    },
    leftContainer: {
        width: '60%',
        paddingRight: 20
    },
    rightContainer: {
        width: '40%',
        flex: '1'
    },
}

LmcHandoverHistoryItem.propTypes = {
    handover: PropTypes.object.isRequired
}