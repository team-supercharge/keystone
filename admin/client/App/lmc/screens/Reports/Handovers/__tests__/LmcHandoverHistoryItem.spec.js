import React from 'react'
import { shallow } from 'enzyme'
import LmcHandoverHistoryItem from '../components/LmcHandoverHistoryItem.jsx'

describe('LmcHandoverHistoryItem', () => {
    let wrapper
    let handover

    beforeEach(() => {
        handover = { 
            notes: [{ id: 'TestId', note: 'This is a note', createdBy: 'TestCarer' }],
            logsByResident: [
                { 
                    logs: [{ id: 'TestId', description: 'This is a log', createdBy: 'TestCarer' }],
                    resident: { id: 'TestId2', name: 'TestResident', picture: 'TestPicture' }
                }
            ],
            seenBy: ['TestId1', 'TestId2', 'TestId3'],
            createdOn: new Date('1/1/2019').toString(),
            createdBy: {},
            witnessedBy: {}
        }
        wrapper = shallow(
            <LmcHandoverHistoryItem
                handover={handover}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders an LmcHandoverTitleBar with the correct props', () => {
        const bar = wrapper.find('LmcHandoverTitleBar')
        const { createdOn, createdBy, witnessedBy } = bar.props()
        expect(createdOn).toEqual(handover.createdOn)
        expect(createdBy).toEqual(handover.createdBy)
        expect(witnessedBy).toEqual(handover.witnessedBy)
    })

    it('renders an LmcHandoverResidentItem with the correct props', () => {
       const residentItem = wrapper.find('LmcHandoverResidentItem')
       expect(residentItem.props().data).toEqual(handover.logsByResident[0]) 
    })

    it('renders an LmcHandoverNotes with the correct props', () => {
        const notes = wrapper.find('LmcHandoverNotes')
        expect(notes.props().notes).toEqual(handover.notes)
    })
})