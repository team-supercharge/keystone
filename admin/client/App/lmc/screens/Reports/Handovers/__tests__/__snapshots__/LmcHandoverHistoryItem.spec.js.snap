// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcHandoverHistoryItem renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <LmcHandoverTitleBar
      createdBy={Object {}}
      createdOn="Tue Jan 01 2019 00:00:00 GMT+0000 (GMT)"
      onClick={[Function]}
      witnessedBy={Object {}}
    />
    <AnimateHeight
      animateOpacity={false}
      animationStateClasses={
        Object {
          "animating": "rah-animating",
          "animatingDown": "rah-animating--down",
          "animatingToHeightAuto": "rah-animating--to-height-auto",
          "animatingToHeightSpecific": "rah-animating--to-height-specific",
          "animatingToHeightZero": "rah-animating--to-height-zero",
          "animatingUp": "rah-animating--up",
          "static": "rah-static",
          "staticHeightAuto": "rah-static--height-auto",
          "staticHeightSpecific": "rah-static--height-specific",
          "staticHeightZero": "rah-static--height-zero",
        }
      }
      applyInlineTransitions={true}
      delay={0}
      duration={500}
      easing="ease"
      height={0}
      style={Object {}}
    >
      <div
        style={
          Object {
            "borderBottom": "1px #eaeaea solid",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 10,
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "paddingRight": 20,
              "width": "60%",
            }
          }
        >
          <div>
            <LmcHandoverResidentItem
              data={
                Object {
                  "logs": Array [
                    Object {
                      "createdBy": "TestCarer",
                      "description": "This is a log",
                      "id": "TestId",
                    },
                  ],
                  "resident": Object {
                    "id": "TestId2",
                    "name": "TestResident",
                    "picture": "TestPicture",
                  },
                }
              }
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flex": "1",
              "flexDirection": "column",
              "width": "40%",
            }
          }
        >
          <LmcHandoverNotes
            notes={
              Array [
                Object {
                  "createdBy": "TestCarer",
                  "id": "TestId",
                  "note": "This is a note",
                },
              ]
            }
          />
          <Connect(LmcSeenByList)
            seenBy={
              Array [
                "TestId1",
                "TestId2",
                "TestId3",
              ]
            }
          />
        </div>
      </div>
    </AnimateHeight>
  </div>,
  "nodes": Array [
    <div>
      <LmcHandoverTitleBar
        createdBy={Object {}}
        createdOn="Tue Jan 01 2019 00:00:00 GMT+0000 (GMT)"
        onClick={[Function]}
        witnessedBy={Object {}}
      />
      <AnimateHeight
        animateOpacity={false}
        animationStateClasses={
          Object {
            "animating": "rah-animating",
            "animatingDown": "rah-animating--down",
            "animatingToHeightAuto": "rah-animating--to-height-auto",
            "animatingToHeightSpecific": "rah-animating--to-height-specific",
            "animatingToHeightZero": "rah-animating--to-height-zero",
            "animatingUp": "rah-animating--up",
            "static": "rah-static",
            "staticHeightAuto": "rah-static--height-auto",
            "staticHeightSpecific": "rah-static--height-specific",
            "staticHeightZero": "rah-static--height-zero",
          }
        }
        applyInlineTransitions={true}
        delay={0}
        duration={500}
        easing="ease"
        height={0}
        style={Object {}}
      >
        <div
          style={
            Object {
              "borderBottom": "1px #eaeaea solid",
              "display": "flex",
              "flexDirection": "row",
              "marginBottom": 10,
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "paddingRight": 20,
                "width": "60%",
              }
            }
          >
            <div>
              <LmcHandoverResidentItem
                data={
                  Object {
                    "logs": Array [
                      Object {
                        "createdBy": "TestCarer",
                        "description": "This is a log",
                        "id": "TestId",
                      },
                    ],
                    "resident": Object {
                      "id": "TestId2",
                      "name": "TestResident",
                      "picture": "TestPicture",
                    },
                  }
                }
              />
            </div>
          </div>
          <div
            style={
              Object {
                "display": "flex",
                "flex": "1",
                "flexDirection": "column",
                "width": "40%",
              }
            }
          >
            <LmcHandoverNotes
              notes={
                Array [
                  Object {
                    "createdBy": "TestCarer",
                    "id": "TestId",
                    "note": "This is a note",
                  },
                ]
              }
            />
            <Connect(LmcSeenByList)
              seenBy={
                Array [
                  "TestId1",
                  "TestId2",
                  "TestId3",
                ]
              }
            />
          </div>
        </div>
      </AnimateHeight>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcHandoverHistoryItem
        handover={
          Object {
            "createdBy": Object {},
            "createdOn": "Tue Jan 01 2019 00:00:00 GMT+0000 (GMT)",
            "logsByResident": Array [
              Object {
                "logs": Array [
                  Object {
                    "createdBy": "TestCarer",
                    "description": "This is a log",
                    "id": "TestId",
                  },
                ],
                "resident": Object {
                  "id": "TestId2",
                  "name": "TestResident",
                  "picture": "TestPicture",
                },
              },
            ],
            "notes": Array [
              Object {
                "createdBy": "TestCarer",
                "id": "TestId",
                "note": "This is a note",
              },
            ],
            "seenBy": Array [
              "TestId1",
              "TestId2",
              "TestId3",
            ],
            "witnessedBy": Object {},
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcHandoverHistoryItem {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handover": Object {
            "createdBy": Object {},
            "createdOn": "Tue Jan 01 2019 00:00:00 GMT+0000 (GMT)",
            "logsByResident": Array [
              Object {
                "logs": Array [
                  Object {
                    "createdBy": "TestCarer",
                    "description": "This is a log",
                    "id": "TestId",
                  },
                ],
                "resident": Object {
                  "id": "TestId2",
                  "name": "TestResident",
                  "picture": "TestPicture",
                },
              },
            ],
            "notes": Array [
              Object {
                "createdBy": "TestCarer",
                "id": "TestId",
                "note": "This is a note",
              },
            ],
            "seenBy": Array [
              "TestId1",
              "TestId2",
              "TestId3",
            ],
            "witnessedBy": Object {},
          },
        },
        "refs": Object {},
        "renderData": [Function],
        "renderHandoverCarers": [Function],
        "state": Object {
          "isShowingContent": false,
        },
        "toggleContent": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <LmcHandoverTitleBar
            createdBy={Object {}}
            createdOn="Tue Jan 01 2019 00:00:00 GMT+0000 (GMT)"
            onClick={[Function]}
            witnessedBy={Object {}}
          />
          <AnimateHeight
            animateOpacity={false}
            animationStateClasses={
              Object {
                "animating": "rah-animating",
                "animatingDown": "rah-animating--down",
                "animatingToHeightAuto": "rah-animating--to-height-auto",
                "animatingToHeightSpecific": "rah-animating--to-height-specific",
                "animatingToHeightZero": "rah-animating--to-height-zero",
                "animatingUp": "rah-animating--up",
                "static": "rah-static",
                "staticHeightAuto": "rah-static--height-auto",
                "staticHeightSpecific": "rah-static--height-specific",
                "staticHeightZero": "rah-static--height-zero",
              }
            }
            applyInlineTransitions={true}
            delay={0}
            duration={500}
            easing="ease"
            height={0}
            style={Object {}}
          >
            <div
              style={
                Object {
                  "borderBottom": "1px #eaeaea solid",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginBottom": 10,
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "paddingRight": 20,
                    "width": "60%",
                  }
                }
              >
                <div>
                  <LmcHandoverResidentItem
                    data={
                      Object {
                        "logs": Array [
                          Object {
                            "createdBy": "TestCarer",
                            "description": "This is a log",
                            "id": "TestId",
                          },
                        ],
                        "resident": Object {
                          "id": "TestId2",
                          "name": "TestResident",
                          "picture": "TestPicture",
                        },
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "flex": "1",
                    "flexDirection": "column",
                    "width": "40%",
                  }
                }
              >
                <LmcHandoverNotes
                  notes={
                    Array [
                      Object {
                        "createdBy": "TestCarer",
                        "id": "TestId",
                        "note": "This is a note",
                      },
                    ]
                  }
                />
                <Connect(LmcSeenByList)
                  seenBy={
                    Array [
                      "TestId1",
                      "TestId2",
                      "TestId3",
                    ]
                  }
                />
              </div>
            </div>
          </AnimateHeight>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <LmcHandoverTitleBar
            createdBy={Object {}}
            createdOn="Tue Jan 01 2019 00:00:00 GMT+0000 (GMT)"
            onClick={[Function]}
            witnessedBy={Object {}}
          />
          <AnimateHeight
            animateOpacity={false}
            animationStateClasses={
              Object {
                "animating": "rah-animating",
                "animatingDown": "rah-animating--down",
                "animatingToHeightAuto": "rah-animating--to-height-auto",
                "animatingToHeightSpecific": "rah-animating--to-height-specific",
                "animatingToHeightZero": "rah-animating--to-height-zero",
                "animatingUp": "rah-animating--up",
                "static": "rah-static",
                "staticHeightAuto": "rah-static--height-auto",
                "staticHeightSpecific": "rah-static--height-specific",
                "staticHeightZero": "rah-static--height-zero",
              }
            }
            applyInlineTransitions={true}
            delay={0}
            duration={500}
            easing="ease"
            height={0}
            style={Object {}}
          >
            <div
              style={
                Object {
                  "borderBottom": "1px #eaeaea solid",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginBottom": 10,
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "paddingRight": 20,
                    "width": "60%",
                  }
                }
              >
                <div>
                  <LmcHandoverResidentItem
                    data={
                      Object {
                        "logs": Array [
                          Object {
                            "createdBy": "TestCarer",
                            "description": "This is a log",
                            "id": "TestId",
                          },
                        ],
                        "resident": Object {
                          "id": "TestId2",
                          "name": "TestResident",
                          "picture": "TestPicture",
                        },
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "flex": "1",
                    "flexDirection": "column",
                    "width": "40%",
                  }
                }
              >
                <LmcHandoverNotes
                  notes={
                    Array [
                      Object {
                        "createdBy": "TestCarer",
                        "id": "TestId",
                        "note": "This is a note",
                      },
                    ]
                  }
                />
                <Connect(LmcSeenByList)
                  seenBy={
                    Array [
                      "TestId1",
                      "TestId2",
                      "TestId3",
                    ]
                  }
                />
              </div>
            </div>
          </AnimateHeight>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcHandoverHistoryItem
    handover={
      Object {
        "createdBy": Object {},
        "createdOn": "Tue Jan 01 2019 00:00:00 GMT+0000 (GMT)",
        "logsByResident": Array [
          Object {
            "logs": Array [
              Object {
                "createdBy": "TestCarer",
                "description": "This is a log",
                "id": "TestId",
              },
            ],
            "resident": Object {
              "id": "TestId2",
              "name": "TestResident",
              "picture": "TestPicture",
            },
          },
        ],
        "notes": Array [
          Object {
            "createdBy": "TestCarer",
            "id": "TestId",
            "note": "This is a note",
          },
        ],
        "seenBy": Array [
          "TestId1",
          "TestId2",
          "TestId3",
        ],
        "witnessedBy": Object {},
      }
    }
  />,
}
`;
